batch/batch/server/server.py

```diff
@@ -492,6 +493,7 @@ def is_successful(self):
                 traceback.print_tb(err.__traceback__)
                 log.info(f'ignoring pod deletion failure for job {self.id} due to {err}')
             self._pod_name = None
+        await self.set_state('Ready')
         await self._create_pod()
```

Comment:
This is not really related to PR, so if you prefer we can skip; I think you should try to have as few updates to the db as possible, mostly because doing so will help us reduce the number of ways that we can get out-of-sync updates (database update races). In `_create_pod` I see:

```python
await db.jobs.update_record(*self.id,
                                    pod_name=self._pod_name)
        await self.set_state('Running')
```

This I think could be 

```python
 await db.jobs.update_record(*self.id,
                                    pod_name=self._pod_name, state='Running')
```

Which will remove O(N) calls, but more importantly remove the possibility than 1 future gets a view that is "Ready" and one "Running", when the true state is "Running".

Reply:
Again, not really this PR, thought I would bring it up in case it were helpful.

Reply:
`mark_unscheduled` can be called when something happens to a running pod such as it was preempted. Therefore, you want to reset the state to Ready before calling create_pod in case that fails. So I think having two calls here is necessary.